diff --git a/patches/src/main/kotlin/app/morphe/patches/music/misc/playservice/VersionCheckPatch.kt b/patches/src/main/kotlin/app/morphe/patches/music/misc/playservice/VersionCheckPatch.kt
--- a/patches/src/main/kotlin/app/morphe/patches/music/misc/playservice/VersionCheckPatch.kt
+++ b/patches/src/main/kotlin/app/morphe/patches/music/misc/playservice/VersionCheckPatch.kt
@@ -35 +35 @@ val versionCheckPatch = bytecodePatch {
-        val versionName = packageMetadata.versionName
+        val versionName = packageMetadata.versionName.substringBefore("-patches-")
diff --git a/patches/src/main/kotlin/app/morphe/patches/youtube/misc/playservice/VersionCheckPatch.kt b/patches/src/main/kotlin/app/morphe/patches/youtube/misc/playservice/VersionCheckPatch.kt
--- a/patches/src/main/kotlin/app/morphe/patches/youtube/misc/playservice/VersionCheckPatch.kt
+++ b/patches/src/main/kotlin/app/morphe/patches/youtube/misc/playservice/VersionCheckPatch.kt
@@ -77 +77 @@ val versionCheckPatch = bytecodePatch {
-        val versionName = packageMetadata.versionName
+        val versionName = packageMetadata.versionName.substringBefore("-patches-")
