--- a/lib/vector_math_64.dart
+++ b/lib/vector_math_64.dart
@@ -1,3 +1,9 @@
 library vector_math_64;
 
 part "src/vector_math_64/matrix4.dart";
+
+extension Matrix4CompatMethods on Matrix4 {
+  Matrix4 scaleByDouble(double x, double y, double z, double w) => scale(x, y, z);
+  Matrix4 translateByDouble(double x, double y, double z, double w) => translate(x, y, z);
+  Matrix4 leftTranslateByDouble(double x, double y, double z, double w) => leftTranslate(x, y, z);
+}
