--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -24,8 +24,6 @@
     alias(libs.plugins.compose.multiplatform) apply false
     alias(libs.plugins.datadog) apply false
     alias(libs.plugins.devtools.ksp) apply false
-    alias(libs.plugins.firebase.crashlytics) apply false
-    alias(libs.plugins.firebase.perf) apply false
     alias(libs.plugins.google.services) apply false
     alias(libs.plugins.hilt) apply false
     alias(libs.plugins.room) apply false
