--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -20,13 +20,10 @@
     repositories {
         gradlePluginPortal()
         maven { url = uri("https://repo.gradle.org/gradle/libs-releases") }
+        mavenCentral()
     }
 }

-plugins {
-    id("com.gradle.enterprise").version("3.5")
-    id("com.gradle.enterprise.gradle-enterprise-conventions-plugin").version("0.7.2")
-}

 apply(from = "gradle/shared-with-buildSrc/mirrors.settings.gradle.kts")
