--- a/subprojects/internal-integ-testing/build.gradle.kts
+++ b/subprojects/internal-integ-testing/build.gradle.kts
@@ -109,12 +109,12 @@
 }
 
 val prepareVersionsInfo = tasks.register<PrepareVersionsInfo>("prepareVersionsInfo") {
-    destFile = layout.buildDirectory.file("generated-resources/all-released-versions/all-released-versions.properties")
-    versions = moduleIdentity.releasedVersions.map {
+    destFile.set(layout.buildDirectory.file("generated-resources/all-released-versions/all-released-versions.properties"))
+    versions.set(moduleIdentity.releasedVersions.map {
         it.allPreviousVersions.joinToString(" ") { it.version }
-    }
-    mostRecent = moduleIdentity.releasedVersions.map { it.mostRecentRelease.version }
-    mostRecentSnapshot = moduleIdentity.releasedVersions.map { it.mostRecentSnapshot.version }
+    })
+    mostRecent.set(moduleIdentity.releasedVersions.map { it.mostRecentRelease.version })
+    mostRecentSnapshot.set(moduleIdentity.releasedVersions.map { it.mostRecentSnapshot.version })
 }
 
 val copyAgpVersionsInfo by tasks.registering(Copy::class) {
@@ -124,8 +124,8 @@
 
 val generateLanguageAnnotations by tasks.registering(GenerateLanguageAnnotations::class) {
     classpath.from(configurations.integTestDistributionRuntimeClasspath)
-    packageName = "org.gradle.integtests.fixtures"
-    destDir = layout.buildDirectory.dir("generated/sources/language-annotations/groovy/main")
+    packageName.set("org.gradle.integtests.fixtures")
+    destDir.set(layout.buildDirectory.dir("generated/sources/language-annotations/groovy/main"))
 }
 
 sourceSets.main {
