--- a/build.gradle
+++ b/build.gradle
@@ -24,10 +24,7 @@
         classpath libs.gradle.gradlePlugin
         classpath libs.gradle.kotlinPlugin
         classpath libs.gradle.hiltPlugin
-        classpath 'com.google.firebase:firebase-appdistribution-gradle:4.0.0'
-        classpath 'com.google.gms:google-services:4.3.15'
         classpath 'org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:4.0.0.2929'
-        classpath 'com.google.android.gms:oss-licenses-plugin:0.10.6'
         classpath "com.likethesalad.android:stem-plugin:2.9.0"
         classpath 'org.owasp:dependency-check-gradle:8.2.1'
         classpath "org.jetbrains.dokka:dokka-gradle-plugin:1.8.10"
