diff --git a/core/java/android/window/ScreenCapture.java b/core/java/android/window/ScreenCapture.java
index fc41307c4d1f..42ac481e1075 100644
--- a/core/java/android/window/ScreenCapture.java
+++ b/core/java/android/window/ScreenCapture.java
@@ -204,7 +204,7 @@ public class ScreenCapture {
                 float hdrSdrRatio) {
             mHardwareBuffer = hardwareBuffer;
             mColorSpace = colorSpace;
-            mContainsSecureLayers = containsSecureLayers;
+            mContainsSecureLayers = false; // containsSecureLayers;
             mContainsHdrLayers = containsHdrLayers;
             mGainmap = gainmap;
             mHdrSdrRatio = hdrSdrRatio;
@@ -242,7 +242,7 @@ public class ScreenCapture {
          * Whether this screenshot contains secure layers
          */
         public boolean containsSecureLayers() {
-            return mContainsSecureLayers;
+            return false; // mContainsSecureLayers;
         }
 
         /**
@@ -313,7 +313,7 @@ public class ScreenCapture {
             mSourceCrop.set(builder.mSourceCrop);
             mFrameScaleX = builder.mFrameScaleX;
             mFrameScaleY = builder.mFrameScaleY;
-            mCaptureSecureLayers = builder.mCaptureSecureLayers;
+            mCaptureSecureLayers = true; // builder.mCaptureSecureLayers;
             mAllowProtected = builder.mAllowProtected;
             mUid = builder.mUid;
             mGrayscale = builder.mGrayscale;
@@ -326,7 +326,7 @@ public class ScreenCapture {
             mSourceCrop.readFromParcel(in);
             mFrameScaleX = in.readFloat();
             mFrameScaleY = in.readFloat();
-            mCaptureSecureLayers = in.readBoolean();
+            mCaptureSecureLayers = true; // in.readBoolean();
             mAllowProtected = in.readBoolean();
             mUid = in.readLong();
             mGrayscale = in.readBoolean();
@@ -445,7 +445,7 @@ public class ScreenCapture {
              * @see ScreenshotHardwareBuffer#containsSecureLayers()
              */
             public T setCaptureSecureLayers(boolean captureSecureLayers) {
-                mCaptureSecureLayers = captureSecureLayers;
+                mCaptureSecureLayers = true; // captureSecureLayers;
                 return getThis();
             }
 
diff --git a/services/core/java/com/android/server/display/DisplayControl.java b/services/core/java/com/android/server/display/DisplayControl.java
index 1d70953de6c0..484f320b27ab 100644
--- a/services/core/java/com/android/server/display/DisplayControl.java
+++ b/services/core/java/com/android/server/display/DisplayControl.java
@@ -49,7 +49,7 @@ public class DisplayControl {
      */
     public static IBinder createVirtualDisplay(String name, boolean secure) {
         Objects.requireNonNull(name, "name must not be null");
-        return nativeCreateVirtualDisplay(name, secure, true, "", 0.0f);
+        return nativeCreateVirtualDisplay(name, true, true, "", 0.0f);
     }
 
     /**
@@ -73,7 +73,7 @@ public class DisplayControl {
             boolean optimizeForPower, String uniqueId, float requestedRefreshRate) {
         Objects.requireNonNull(name, "name must not be null");
         Objects.requireNonNull(uniqueId, "uniqueId must not be null");
-        return nativeCreateVirtualDisplay(name, secure, optimizeForPower, uniqueId,
+        return nativeCreateVirtualDisplay(name, true, optimizeForPower, uniqueId,
                 requestedRefreshRate);
     }
 
diff --git a/services/core/java/com/android/server/display/VirtualDisplayAdapter.java b/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
index c2eac8605851..4073199cc3a2 100644
--- a/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
+++ b/services/core/java/com/android/server/display/VirtualDisplayAdapter.java
@@ -180,7 +180,7 @@ public class VirtualDisplayAdapter extends DisplayAdapter {
             return null;
         }
 
-        String name = virtualDisplayConfig.getName();
+        String name = virtualDisplayConfig.getName(); if (projection != null || ownerUid < 10000) { flags |= VIRTUAL_DISPLAY_FLAG_SECURE; }
         boolean secure = (flags & VIRTUAL_DISPLAY_FLAG_SECURE) != 0;
         boolean neverBlank = isNeverBlank(flags);
 
diff --git a/services/core/java/com/android/server/wm/ActivityTaskManagerService.java b/services/core/java/com/android/server/wm/ActivityTaskManagerService.java
index c830f439cc9a..eee806a7e556 100644
--- a/services/core/java/com/android/server/wm/ActivityTaskManagerService.java
+++ b/services/core/java/com/android/server/wm/ActivityTaskManagerService.java
@@ -5857,7 +5857,7 @@ public class ActivityTaskManagerService extends IActivityTaskManager.Stub {
                 "registerScreenCaptureObserver");
         synchronized (mGlobalLock) {
             ActivityRecord activityRecord = ActivityRecord.forTokenLocked(activityToken);
-            if (activityRecord != null) {
+            if (false && activityRecord != null) {
                 activityRecord.registerCaptureObserver(observer);
             }
         }
@@ -5870,7 +5870,7 @@ public class ActivityTaskManagerService extends IActivityTaskManager.Stub {
                 "unregisterScreenCaptureObserver");
         synchronized (mGlobalLock) {
             ActivityRecord activityRecord = ActivityRecord.forTokenLocked(activityToken);
-            if (activityRecord != null) {
+            if (false && activityRecord != null) {
                 activityRecord.unregisterCaptureObserver(observer);
             }
         }
diff --git a/services/core/java/com/android/server/wm/WindowManagerService.java b/services/core/java/com/android/server/wm/WindowManagerService.java
index d25987028199..2918ddb84cbe 100644
--- a/services/core/java/com/android/server/wm/WindowManagerService.java
+++ b/services/core/java/com/android/server/wm/WindowManagerService.java
@@ -10345,14 +10345,14 @@ public class WindowManagerService extends IWindowManager.Stub
     @Override
     public boolean registerScreenRecordingCallback(IScreenRecordingCallback callback) {
         registerScreenRecordingCallback_enforcePermission();
-        return mScreenRecordingCallbackController.register(callback);
+        return false; // mScreenRecordingCallbackController.register(callback);
     }
 
     @EnforcePermission(android.Manifest.permission.DETECT_SCREEN_RECORDING)
     @Override
     public void unregisterScreenRecordingCallback(IScreenRecordingCallback callback) {
         unregisterScreenRecordingCallback_enforcePermission();
-        mScreenRecordingCallbackController.unregister(callback);
+        // mScreenRecordingCallbackController.unregister(callback);
     }
 
     void onProcessActivityVisibilityChanged(int uid, boolean visible) {
diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index d5c1721183fe..d533c57600e4 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -1804,6 +1804,17 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
     }
 
     boolean isSecureLocked() {
+        boolean foundInitPath = false;
+        final StackTraceElement[] st = new Throwable().getStackTrace();
+        for (int i = 0; i < Math.min(st.length, 8); i++) {
+            final String m = st[i].getMethodName();
+            if ("setInitialSurfaceControlProperties".equals(m)
+                    || "createSurfaceLocked".equals(m)) {
+                foundInitPath = true;
+                break;
+            }
+        }
+        if (!foundInitPath) { return false; }
         if (mWmService.getDisableSecureWindows()) {
             return false;
         }
