diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index d5c1721183fe..47bdad317aea 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -1804,7 +1804,7 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
     }
 
     boolean isSecureLocked() {
-        if (mWmService.getDisableSecureWindows()) {
+        if (true || mWmService.getDisableSecureWindows()) {
             return false;
         }
 
