--- a/buildSrc/src/main/kotlin/breezy/buildlogic/ProjectConfiguration.kt
+++ b/buildSrc/src/main/kotlin/breezy/buildlogic/ProjectConfiguration.kt
@@ -10,7 +10,7 @@
 object ProjectConfiguration {
     const val MIN_SDK = 26
     const val TARGET_SDK = 35
-    const val COMPILE_SDK = 37
+    const val COMPILE_SDK = 36
     const val COMPILE_SDK_PREVIEW = "VanillaIceCream"
     const val BUILD_TOOLS_VERSION = "36.0.0"
 
