--- settings.gradle.kts
+++ settings.gradle.kts
@@ -5,14 +5,9 @@
         mavenLocal()
         gradlePluginPortal()
         google()
-        maven {
-            name = "GitHubPackages"
-            url = uri("https://maven.pkg.github.com/MorpheApp/registry")
-            credentials {
-                username = providers.gradleProperty("gpr.user").getOrElse(System.getenv("GITHUB_ACTOR"))
-                password = providers.gradleProperty("gpr.key").getOrElse(System.getenv("GITHUB_TOKEN"))
-            }
-        }
+        maven { url = uri("file://" + System.getenv("MORPHE_PLUGIN_M2")) }
+        maven { url = uri("file://" + System.getenv("MORPHE_LIBRARY_M2")) }
+        maven { url = uri("file://" + System.getenv("MORPHE_PATCHES_LIBRARY_M2")) }
         // Obtain baksmali/smali from source builds - https://github.com/iBotPeaches/smali
         // Remove when official smali releases come out again.
         maven { url = uri("https://jitpack.io") }
