--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -23,24 +23,7 @@
     }
 }
 
-val commitCount = run {
-    val repo = FileRepository(rootProject.file(".git"))
-    val refId = repo.refDatabase.exactRef("refs/remotes/origin/master").objectId!!
-    Git(repo).log().add(refId).call().count()
-}
-
-val (coreCommitCount, coreLatestTag) = FileRepositoryBuilder().setGitDir(rootProject.file(".git/modules/core"))
-    .runCatching {
-        build().use { repo ->
-            val git = Git(repo)
-            val coreCommitCount =
-                git.log()
-                    .add(repo.refDatabase.exactRef("HEAD").objectId)
-                    .call().count()
-            val ver = git.describe()
-                .setTags(true)
-                .setAbbrev(0).call().removePrefix("v")
-            coreCommitCount to ver
-        }
-    }.getOrNull() ?: (1145 to "1.0")
+val commitCount = 20260315
+val coreCommitCount = 7213
+val coreLatestTag = "1.11.0"
 
