diff -urN rain-src/android/app/build.gradle rain-src-3/android/app/build.gradle
--- rain-src/android/app/build.gradle	2026-06-05 09:53:23.507783788 +1000
+++ rain-src-3/android/app/build.gradle	2026-06-05 10:09:47.229996331 +1000
@@ -66,7 +66,8 @@
 
     buildTypes {
         release {
-           signingConfig = signingConfigs.release
+           signingConfig = signingConfigs.debug
+           proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
         }
 
         debug {
diff -urN rain-src/android/app/proguard-rules.pro rain-src-3/android/app/proguard-rules.pro
--- rain-src/android/app/proguard-rules.pro	1970-01-01 10:00:00.000000000 +1000
+++ rain-src-3/android/app/proguard-rules.pro	2026-06-05 10:09:47.229996331 +1000
@@ -0,0 +1,4 @@
+-dontwarn com.google.android.gms.**
+-dontwarn com.google.android.gms.location.**
+-dontwarn com.google.android.gms.common.**
+-dontwarn com.google.android.gms.tasks.**
diff -urN rain-src/android/build.gradle rain-src-3/android/build.gradle
--- rain-src/android/build.gradle	2026-06-05 09:53:23.507783788 +1000
+++ rain-src-3/android/build.gradle	2026-06-05 10:09:47.229996331 +1000
@@ -16,3 +16,13 @@
 tasks.register("clean", Delete) {
     delete rootProject.buildDir
 }
+
+subprojects {
+    project.configurations.all {
+        resolutionStrategy.eachDependency { details ->
+            if (details.requested.group == 'androidx.glance' && details.requested.name == 'glance-appwidget') {
+                details.useVersion '1.1.1'
+            }
+        }
+    }
+}
