--- build.gradle.kts
+++ build.gradle.kts
@@ -6,9 +6,7 @@
-    alias(libs.plugins.binary.compatibility.validator)
     `maven-publish`
-    signing
-    jacoco
 }
 
@@ -50,14 +48,7 @@
-    maven {
-        // A repository must be specified for some reason. "registry" is a dummy.
-        url = uri("https://maven.pkg.github.com/MorpheApp/registry")
-        credentials {
-            username = project.findProperty("gpr.user") as String? ?: System.getenv("GITHUB_ACTOR")
-            password = project.findProperty("gpr.key") as String? ?: System.getenv("GITHUB_TOKEN")
-        }
-    }
+    maven { url = uri("file://" + System.getenv("MORPHE_PLUGIN_M2")) }
 }
 
@@ -104,17 +95,7 @@
 publishing {
     repositories {
-        maven {
-            name = "GitHubPackages"
-            url = uri("https://maven.pkg.github.com/MorpheApp/morphe-patcher")
-            credentials {
-                username = System.getenv("GITHUB_ACTOR")
-                password = System.getenv("GITHUB_TOKEN")
-            }
-        }
+        maven { url = uri("file://" + project.projectDir.resolve("build/m2").absolutePath) }
     }
-
     publications {
@@ -151,5 +132,4 @@
-signing {
-    useGpgCmd()
-    sign(publishing.publications["morphe-patcher-publication"])
-}
+tasks.withType<Sign> { enabled = false }
