diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
index c7bfd7b6c2f..c7685f7d642 100644
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -1,13 +1,7 @@
-import org.apache.ivy.core.install.InstallOptions
-import org.apache.ivy.core.module.descriptor.DefaultModuleDescriptor
-import org.apache.ivy.core.module.descriptor.DependencyDescriptor
-import org.gradle.api.DependencyManager
-
 usePlugin('groovy')
 
 dependencies {
     addFlatDirResolver('lib', new File(rootDir, 'lib'))
-    dependencies(['compile'], ":svnkit:1.1.6", ":svnkit-javahl:1.1.6", ":xmlunit:1.2")
 }
 
 sourceCompatibility = 1.5
@@ -23,14 +17,3 @@ test {
     // We set forkmode to ONCE as our tests are written in Groovy and the startup time of Groovy is significant.
     options.fork(forkMode: org.gradle.api.tasks.testing.ForkMode.ONCE)
 }
-
-uploadLibs.doLast {
-    InstallOptions installOptions = new InstallOptions()
-    installOptions.overwrite = true
-    installOptions.validate = false
-    def ivy = dependencies.ivy
-    DefaultModuleDescriptor moduleDescriptor = dependencies.moduleDescriptorConverter.convert(dependencies, true)
-    moduleDescriptor.dependencies.each { DependencyDescriptor dependencyDescriptor ->
-        ivy.install(dependencyDescriptor.dependencyRevisionId, 'lib', DependencyManager.BUILD_RESOLVER_NAME, installOptions)
-    }
-}
\ No newline at end of file
diff --git a/buildSrc/src/main/groovy/org/gradle/build/integtests/JavaProject.groovy b/buildSrc/src/main/groovy/org/gradle/build/integtests/JavaProject.groovy
index 7401073cf7b..ce197cffe42 100644
--- a/buildSrc/src/main/groovy/org/gradle/build/integtests/JavaProject.groovy
+++ b/buildSrc/src/main/groovy/org/gradle/build/integtests/JavaProject.groovy
@@ -17,9 +17,6 @@
 package org.gradle.build.integtests
 
 import groovy.text.SimpleTemplateEngine
-import org.custommonkey.xmlunit.Diff
-import org.custommonkey.xmlunit.ElementNameAndAttributeQualifier
-import org.custommonkey.xmlunit.XMLAssert
 import static org.junit.Assert.assertTrue
 
 
@@ -112,9 +109,11 @@ class JavaProject {
     }
 
     private static void compareXmlWithIgnoringOrder(String expectedXml, String actualXml) {
-        Diff diff = new Diff(expectedXml, actualXml)
-        diff.overrideElementQualifier(new ElementNameAndAttributeQualifier())
-        XMLAssert.assertXMLEqual(diff, true);
+        assert normalizeXml(expectedXml) == normalizeXml(actualXml)
+    }
+
+    private static String normalizeXml(String xml) {
+        xml.replaceAll(">\\s+<", "><").trim()
     }
 
     private static String replaceWithGradleHome(String resourcePath, String gradleHome) {
diff --git a/buildSrc/src/main/groovy/org/gradle/build/release/Svn.groovy b/buildSrc/src/main/groovy/org/gradle/build/release/Svn.groovy
index 7e636044be9..079a6b8815f 100644
--- a/buildSrc/src/main/groovy/org/gradle/build/release/Svn.groovy
+++ b/buildSrc/src/main/groovy/org/gradle/build/release/Svn.groovy
@@ -18,165 +18,24 @@ package org.gradle.build.release
 
 import org.gradle.api.GradleException
 import org.gradle.api.Project
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
-import org.tigris.subversion.javahl.Revision
-import org.tigris.subversion.javahl.Status
-import org.tmatesoft.svn.core.SVNException
-import org.tmatesoft.svn.core.SVNURL
-import org.tmatesoft.svn.core.internal.io.dav.DAVRepositoryFactory
-import org.tmatesoft.svn.core.internal.util.SVNPathUtil
-import org.tmatesoft.svn.core.javahl.SVNClientImpl
-import org.tmatesoft.svn.core.wc.SVNClientManager
-import org.tmatesoft.svn.core.wc.SVNRevision
-import org.tmatesoft.svn.core.wc.SVNStatusClient
-import org.tmatesoft.svn.core.wc.SVNWCUtil
 
 /**
  * @author Hans Dockter
  */
 class Svn {
-    private static Logger logger = LoggerFactory.getLogger(Svn)
-
     Project project
-    SVNClientManager clientManager
-    SVNStatusClient statusClient
-    SVNClientImpl javaHlClient
-    boolean alwaysTrunk = false
 
     Svn() {}
 
     Svn(Project project) {
-        assert project
         this.project = project
-        try {
-            DAVRepositoryFactory.setup();
-            clientManager = SVNClientManager.newInstance(
-                    SVNWCUtil.createDefaultOptions(true), project.codehausUserName, project.codehausUserPassword)
-            statusClient = clientManager.getStatusClient()
-            javaHlClient = SVNClientImpl.newInstance()
-            javaHlClient.username(project.codehausUserName)
-            javaHlClient.password(project.codehausUserPassword)
-            throwExceptionIfNoSvnProject()
-        } catch (Throwable e) {
-            logger.info("""Can't access svn working copy. Maybe this is not an svn project or the codehausUserName/password property is not set.
-                Releasing won't be possible. It is assumed that this isTrunk is true.""")
-            alwaysTrunk = true
-        }
-    }
-
-    void throwExceptionIfNoSvnProject() {
-        svnDir
-    }
-
-    def release() {
-        checkWorkingCopy()
-        if (isTrunk() || project.version.toString() == branchVersion) {
-            majorOrMinorRelease()
-        } else {
-            revisionRelease()
-        }
-    }
-
-    def majorOrMinorRelease() {
-        if (isTrunk()) {
-            exitIfReleaseBranchDirectoryExists()
-        }
-        project.version.storeCurrentVersion()
-        commitProperties()
-        if (isTrunk()) {
-            copyTrunkToReleaseBranch()
-        }
-        tagReleaseBranch()
-    }
-
-    def revisionRelease() {
-        project.version.storeCurrentVersion()
-        commitProperties()
-        tagReleaseBranch()
-    }
-
-
-    def commitProperties() {
-        File props = new File(project.projectDir, 'gradle.properties')
-        javaHlClient.commit([props.absolutePath] as String[], "Incremented version properties", false)
-    }
-
-    def exitIfReleaseBranchDirectoryExists() {
-        try {
-            clientManager.WCClient.doInfo(new SVNURL(releaseBranchUrl, true),
-                    SVNRevision.UNDEFINED, SVNRevision.HEAD)
-            throw new GradleException("Release branch directory $releaseBranchUrl already exists. You can't release from trunk therefore.")
-        } catch (SVNException ignore) {
-            // SVNException means directory does not exists. Which is what we want.
-        }
-    }
-
-    def copyTrunkToReleaseBranch() {
-        javaHlClient.copy(trunkUrl, releaseBranchUrl, "Copy trunk to release branch $releaseBranchName" as String, Revision.HEAD)
-    }
-
-    def tagReleaseBranch() {
-        javaHlClient.copy(releaseBranchUrl, tagsUrl, "Tag release $releaseTagName" as String, Revision.HEAD)
-    }
-
-    def getReleaseBranchName() {
-        'RB-' + majorMinorVersion
-    }
-
-    def getReleaseTagName() {
-        'REL-' + project.version
-    }
-
-    def getMajorMinorVersion() {
-        project.version.toString().split('\\.')[0..1].join('.')
-    }
-
-    def getReleaseBranchUrl() {
-        svnBaseUrl + "/branches/$releaseBranchName"
-    }
-
-    def getTagsUrl() {
-        svnBaseUrl + "/tags/$releaseTagName"
-    }
-
-    def getTrunkUrl() {
-        svnBaseUrl + "/trunk"
-    }
-
-    String getSvnUrl() {
-        statusClient.doStatus(project.projectDir, false).URL
-    }
-
-    String getSvnBaseUrl() {
-        String url = svnUrl
-        url = SVNPathUtil.removeTail(url)
-        if (!isTrunk()) {
-            url = SVNPathUtil.removeTail(url)
-        }
-        url
     }
 
     boolean isTrunk() {
-        alwaysTrunk || svnDir == 'trunk'
+        true
     }
 
-    String getSvnDir() {
-        svnUrl.substring(svnUrl.lastIndexOf('/') + 1)
-    }
-
-    String getBranchVersion() {
-        svnDir.replaceFirst('RB-', '')
-    }
-
-    void checkWorkingCopy() {
-        List statuses = javaHlClient.status(project.projectDir.absolutePath, true, true, false)
-//        ISVNStatusHandler statusHandler = {SVNStatus status -> statuses << status} as ISVNStatusHandler
-//        statusClient.doStatus(project.projectDir, true, true, false, false, statusHandler)
-        if (statuses) {
-            String result = ''
-            statuses.each {Status status -> result += status.path + ' ' + status.url.toString() + '\n'}
-            throw new GradleException("The working copy is not in sync with the SVN server. We can't release!\n$result")
-        }
+    def release() {
+        throw new GradleException("release task is disabled in bootstrap builds")
     }
 }
diff --git a/src/main/groovy/org/gradle/api/dependencies/ArtifactDependency.java b/src/main/groovy/org/gradle/api/dependencies/ArtifactDependency.java
index b1eafc1663c..b7984f2eaf2 100644
--- a/src/main/groovy/org/gradle/api/dependencies/ArtifactDependency.java
+++ b/src/main/groovy/org/gradle/api/dependencies/ArtifactDependency.java
@@ -55,12 +55,10 @@ public class ArtifactDependency extends AbstractDependency {
         DefaultDependencyDescriptor dd = (DefaultDependencyDescriptor) dependencyDescriptorFactory.createDescriptor(parent,
                 (String) groups.get("core"), force, false, false, getConfs(), new ArrayList<ExcludeRule>());
         DefaultDependencyArtifactDescriptor artifactDescriptor = new DefaultDependencyArtifactDescriptor(
-                dd, dd.getDependencyRevisionId().getName(),
-                groups.get("extension"), groups.get("extension"), null, null);
-        dd.addDependencyArtifact("*", artifactDescriptor);
+                dd.getDependencyRevisionId().getName(), groups.get("extension"), groups.get("extension"), null, null);
+        dd.addDependencyArtifact(Dependency.DEFAULT_CONFIGURATION, artifactDescriptor);
         for (String conf : getConfs()) {
             dd.addDependencyConfiguration(conf, Dependency.DEFAULT_CONFIGURATION);
-            artifactDescriptor.addConfiguration(conf);
         }
         return dd;
     }
diff --git a/src/main/groovy/org/gradle/api/internal/dependencies/ResolverFactory.java b/src/main/groovy/org/gradle/api/internal/dependencies/ResolverFactory.java
index 40d7a4d58bf..0ff7fb3a8cb 100644
--- a/src/main/groovy/org/gradle/api/internal/dependencies/ResolverFactory.java
+++ b/src/main/groovy/org/gradle/api/internal/dependencies/ResolverFactory.java
@@ -71,7 +71,7 @@ public class ResolverFactory {
         iBiblioResolver.setM2compatible(true);
         iBiblioResolver.setUseMavenMetadata(true);
         if (jarRepoUrls.length == 0) {
-            iBiblioResolver.setDescriptor(IBiblioResolver.DESCRIPTOR_OPTIONAL);
+            iBiblioResolver.setAllownomd(true);
             return iBiblioResolver;
         }
         iBiblioResolver.setName(iBiblioResolver.getName() + "_poms");
diff --git a/src/main/groovy/org/gradle/api/internal/project/DefaultProject.groovy b/src/main/groovy/org/gradle/api/internal/project/DefaultProject.groovy
index c18757a69d2..58ed61e6f94 100644
--- a/src/main/groovy/org/gradle/api/internal/project/DefaultProject.groovy
+++ b/src/main/groovy/org/gradle/api/internal/project/DefaultProject.groovy
@@ -137,7 +137,7 @@ class DefaultProject extends AbstractProject {
         configureProjects(getAllprojects(), configureClosure);
     }
 
-    public void configureProjects(Iterable<Project> projects, Closure configureClosure) {
+    public void configureProjects(Iterable projects, Closure configureClosure) {
         for (Project project : projects) {
             ConfigureUtil.configure(configureClosure, project);
         }
diff --git a/src/main/groovy/org/gradle/api/tasks/ide/eclipse/EclipseClasspath.java b/src/main/groovy/org/gradle/api/tasks/ide/eclipse/EclipseClasspath.java
index 8a1bdbcaf69..235bf644aad 100644
--- a/src/main/groovy/org/gradle/api/tasks/ide/eclipse/EclipseClasspath.java
+++ b/src/main/groovy/org/gradle/api/tasks/ide/eclipse/EclipseClasspath.java
@@ -127,8 +127,8 @@ public class EclipseClasspath extends ConventionTask {
     }
 
     private void addTestSrcDirs(Element root) {
-        for (String testSrcDir : existingRelativePaths(getTestSrcDirs())) {
-            classpathEntry(root, SRC, testSrcDir).
+        for (Object testSrcDir : existingRelativePaths(getTestSrcDirs())) {
+            classpathEntry(root, SRC, testSrcDir.toString()).
                     addAttribute(OUTPUT, FilenameUtils.separatorsToUnix(relativePath(getTestOutputDirectory())));
         }
     }
