diff --git a/drivers/gpu/mediatek/gpu_mali/mali_valhall/mali-r32p1/drivers/gpu/arm/midgard/context/mali_kbase_context.c b/drivers/gpu/mediatek/gpu_mali/mali_valhall/mali-r32p1/drivers/gpu/arm/midgard/context/mali_kbase_context.c
index 1175d80a2d64..6c20ccc0b102 100644
--- a/drivers/gpu/mediatek/gpu_mali/mali_valhall/mali-r32p1/drivers/gpu/arm/midgard/context/mali_kbase_context.c
+++ b/drivers/gpu/mediatek/gpu_mali/mali_valhall/mali-r32p1/drivers/gpu/arm/midgard/context/mali_kbase_context.c
@@ -145,30 +145,19 @@ int kbase_context_common_init(struct kbase_context *kctx)
 
 	/* Check if this is a Userspace created context */
 	if (likely(kctx->filp)) {
-		struct pid *pid_struct;
 
 		rcu_read_lock();
-		pid_struct = find_get_pid(kctx->tgid);
-		if (likely(pid_struct)) {
-			struct task_struct *task = pid_task(pid_struct, PIDTYPE_PID);
-
-			if (likely(task)) {
-				/* Take a reference on the task to avoid slow lookup
-				 * later on from the page allocation loop.
-				 */
-				get_task_struct(task);
-				kctx->task = task;
-			} else {
-				dev_err(kctx->kbdev->dev,
-					"Failed to get task pointer for %s/%d",
-					current->comm, current->pid);
-				err = -ESRCH;
-			}
-
-			put_pid(pid_struct);
+		struct task_struct *task = current;
+
+		if (likely(task)) {
+			/* Take a reference on the task to avoid slow lookup
+			 * later on from the page allocation loop.
+			 */
+			get_task_struct(task);
+			kctx->task = task;
 		} else {
 			dev_err(kctx->kbdev->dev,
-				"Failed to get pid pointer for %s/%d",
+				"Failed to get task pointer for %s/%d",
 				current->comm, current->pid);
 			err = -ESRCH;
 		}
