--- a/buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt
+++ b/buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt
@@ -194,7 +194,8 @@
                 attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
             }
             extendsFrom(shadedImplementation)
-            outgoing.artifact(shadedJarTask) {
+            outgoing.artifact(shadedJarTask.flatMap { it.jarFile }) {
+                builtBy(shadedJarTask)
                 name = base.archivesBaseName
                 type = "jar"
             }
