--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -369,6 +369,10 @@ dependencies {
     implementation(libs.datastore.preferences)
     implementation(libs.jetbrains.kotlinx.json)
     implementation(libs.kotlin.coroutines)
+    implementation("com.squareup.okio:okio-jvm:3.16.0")
+    implementation("org.apache.commons:commons-lang3:3.18.0")
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:2.2.0")
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json-jvm:1.8.1")
     implementation(libs.timber)
     implementation(libs.zxing)
 
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -12,8 +12,8 @@ espressoCore = "3.6.1" # https://mvnrepository.com/artifact/androidx.test.espres
 feature-delivery = "2.1.0" # https://mvnrepository.com/artifact/com.google.android.play/feature-delivery
 play-integrity = "1.6.0" # https://mvnrepository.com/artifact/com.google.android.play/integrity
 hiltNavigationCompose = "1.2.0" # https://mvnrepository.com/artifact/androidx.hilt/hilt-navigation-compose
-javasteam = "1.8.0.1-21-SNAPSHOT" # https://github.com/joshuatam/JavaSteam/tree/gamenative-latest
-json = "1.8.0" # https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-serialization-json
+javasteam = "1.8.0.1-18-20260407.035245-1" # https://github.com/joshuatam/JavaSteam/tree/gamenative-latest
+json = "1.8.1" # https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-serialization-json
 junit = "4.13.2" # https://mvnrepository.com/artifact/junit/junit
 junitVersion = "1.2.1" # https://mvnrepository.com/artifact/androidx.test.ext/junit
 kotlin = "2.1.21" # https://mvnrepository.com/artifact/org.jetbrains.kotlin.android/org.jetbrains.kotlin.android.gradle.plugin
