diff --git a/config/device/common/sysprop-inclusion.yml b/config/device/common/sysprop-inclusion.yml
index 9666bee3..0ce9bb00 100644
--- a/config/device/common/sysprop-inclusion.yml
+++ b/config/device/common/sysprop-inclusion.yml
@@ -145,9 +145,15 @@ sysprop_inclusions:
   - mmd.zram.comp_algorithm = lz77eh
   - mmd.zram.enabled = true
   - mmd.zram.size = 3221225472
+  - mmd.zram.size = 8589934592
+  - mmd.zram.size = 12884901888
   - mmd.zram.size = 50%
+  - mmd.zram.size = 75%
+  - mmd.zram.size = 100%
   - mmd.zram.writeback.device_size = 1073741824
   - mmd.zram.writeback.device_size = 2147483648
+  - mmd.zram.writeback.device_size = 4294967296
+  - mmd.zram.writeback.device_size = 8589934592
   - mmd.zram.writeback.enabled = true
   - mmd.zram.writeback.huge_idle.enabled = false
   - mmd.zram.writeback.idle.enabled = true
diff --git a/vendor-skels/google_devices/akita/sysprop/product.prop b/vendor-skels/google_devices/akita/sysprop/product.prop
index b2270298..70479662 100644
--- a/vendor-skels/google_devices/akita/sysprop/product.prop
+++ b/vendor-skels/google_devices/akita/sysprop/product.prop
@@ -43,8 +43,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/blazer/sysprop/product.prop b/vendor-skels/google_devices/blazer/sysprop/product.prop
index bc063f59..994faa9d 100644
--- a/vendor-skels/google_devices/blazer/sysprop/product.prop
+++ b/vendor-skels/google_devices/blazer/sysprop/product.prop
@@ -49,8 +49,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=2147483648
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/bluejay/sysprop/product.prop b/vendor-skels/google_devices/bluejay/sysprop/product.prop
index a7971f0e..91ff3ce9 100644
--- a/vendor-skels/google_devices/bluejay/sysprop/product.prop
+++ b/vendor-skels/google_devices/bluejay/sysprop/product.prop
@@ -27,7 +27,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=3221225472
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/caiman/sysprop/product.prop b/vendor-skels/google_devices/caiman/sysprop/product.prop
index a22c8393..1e21cff3 100644
--- a/vendor-skels/google_devices/caiman/sysprop/product.prop
+++ b/vendor-skels/google_devices/caiman/sysprop/product.prop
@@ -47,8 +47,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/cheetah/sysprop/product.prop b/vendor-skels/google_devices/cheetah/sysprop/product.prop
index 51c05c66..e5d89f42 100644
--- a/vendor-skels/google_devices/cheetah/sysprop/product.prop
+++ b/vendor-skels/google_devices/cheetah/sysprop/product.prop
@@ -38,7 +38,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/comet/sysprop/product.prop b/vendor-skels/google_devices/comet/sysprop/product.prop
index 3694f7d0..c696104f 100644
--- a/vendor-skels/google_devices/comet/sysprop/product.prop
+++ b/vendor-skels/google_devices/comet/sysprop/product.prop
@@ -48,8 +48,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/felix/sysprop/product.prop b/vendor-skels/google_devices/felix/sysprop/product.prop
index 5adf4395..4fc4ac47 100644
--- a/vendor-skels/google_devices/felix/sysprop/product.prop
+++ b/vendor-skels/google_devices/felix/sysprop/product.prop
@@ -37,7 +37,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/frankel/sysprop/product.prop b/vendor-skels/google_devices/frankel/sysprop/product.prop
index d8b6ac66..d8cfaa28 100644
--- a/vendor-skels/google_devices/frankel/sysprop/product.prop
+++ b/vendor-skels/google_devices/frankel/sysprop/product.prop
@@ -49,8 +49,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=2147483648
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/husky/sysprop/product.prop b/vendor-skels/google_devices/husky/sysprop/product.prop
index 3613e7b0..f58cd382 100644
--- a/vendor-skels/google_devices/husky/sysprop/product.prop
+++ b/vendor-skels/google_devices/husky/sysprop/product.prop
@@ -44,8 +44,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/komodo/sysprop/product.prop b/vendor-skels/google_devices/komodo/sysprop/product.prop
index 54c64ca0..2a9b1d1d 100644
--- a/vendor-skels/google_devices/komodo/sysprop/product.prop
+++ b/vendor-skels/google_devices/komodo/sysprop/product.prop
@@ -47,8 +47,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/lynx/sysprop/product.prop b/vendor-skels/google_devices/lynx/sysprop/product.prop
index 089314e2..336c6083 100644
--- a/vendor-skels/google_devices/lynx/sysprop/product.prop
+++ b/vendor-skels/google_devices/lynx/sysprop/product.prop
@@ -34,7 +34,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/mustang/sysprop/product.prop b/vendor-skels/google_devices/mustang/sysprop/product.prop
index 80c8fbb4..845e06b8 100644
--- a/vendor-skels/google_devices/mustang/sysprop/product.prop
+++ b/vendor-skels/google_devices/mustang/sysprop/product.prop
@@ -49,8 +49,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=2147483648
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/oriole/sysprop/product.prop b/vendor-skels/google_devices/oriole/sysprop/product.prop
index 460236fe..578bb4e9 100644
--- a/vendor-skels/google_devices/oriole/sysprop/product.prop
+++ b/vendor-skels/google_devices/oriole/sysprop/product.prop
@@ -27,7 +27,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=3221225472
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/panther/sysprop/product.prop b/vendor-skels/google_devices/panther/sysprop/product.prop
index 9d8891d6..a05bf2bb 100644
--- a/vendor-skels/google_devices/panther/sysprop/product.prop
+++ b/vendor-skels/google_devices/panther/sysprop/product.prop
@@ -38,7 +38,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/rango/sysprop/product.prop b/vendor-skels/google_devices/rango/sysprop/product.prop
index ab1c2a57..65e77527 100644
--- a/vendor-skels/google_devices/rango/sysprop/product.prop
+++ b/vendor-skels/google_devices/rango/sysprop/product.prop
@@ -49,8 +49,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=2147483648
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/raven/sysprop/product.prop b/vendor-skels/google_devices/raven/sysprop/product.prop
index ad0905b4..9d9b6a34 100644
--- a/vendor-skels/google_devices/raven/sysprop/product.prop
+++ b/vendor-skels/google_devices/raven/sysprop/product.prop
@@ -27,7 +27,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=3221225472
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/shiba/sysprop/product.prop b/vendor-skels/google_devices/shiba/sysprop/product.prop
index dcb40630..8a4e97d5 100644
--- a/vendor-skels/google_devices/shiba/sysprop/product.prop
+++ b/vendor-skels/google_devices/shiba/sysprop/product.prop
@@ -44,8 +44,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/tangorpro/sysprop/product.prop b/vendor-skels/google_devices/tangorpro/sysprop/product.prop
index e1d7537d..e16c736d 100644
--- a/vendor-skels/google_devices/tangorpro/sysprop/product.prop
+++ b/vendor-skels/google_devices/tangorpro/sysprop/product.prop
@@ -34,7 +34,7 @@ mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
 mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/tegu/sysprop/product.prop b/vendor-skels/google_devices/tegu/sysprop/product.prop
index 9dc53d42..4faab4a9 100644
--- a/vendor-skels/google_devices/tegu/sysprop/product.prop
+++ b/vendor-skels/google_devices/tegu/sysprop/product.prop
@@ -46,8 +46,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
diff --git a/vendor-skels/google_devices/tokay/sysprop/product.prop b/vendor-skels/google_devices/tokay/sysprop/product.prop
index 851d8e4f..5191f677 100644
--- a/vendor-skels/google_devices/tokay/sysprop/product.prop
+++ b/vendor-skels/google_devices/tokay/sysprop/product.prop
@@ -47,8 +47,8 @@ mm.zram.maintenance.first_delay_seconds=86400
 mm.zram.maintenance.periodic_delay_seconds=86400
 mmd.zram.comp_algorithm=lz77eh
 mmd.zram.enabled=true
-mmd.zram.size=50%
-mmd.zram.writeback.device_size=1073741824
+mmd.zram.size=100%
+mmd.zram.writeback.device_size=4294967296
 mmd.zram.writeback.enabled=true
 mmd.zram.writeback.huge_idle.enabled=false
 mmd.zram.writeback.idle.enabled=true
