--- a/private/property_contexts
+++ b/private/property_contexts
@@ -1751,6 +1751,7 @@

 sys.state_of_default_display u:object_r:exported_system_prop:s0 exact string
 persist.sys.alt_touch_mode u:object_r:exported_system_prop:s0 exact string
+persist.sys.husky_high_emission_frequency_mode u:object_r:exported_system_prop:s0 exact string

 # properties for the virtual Face HAL
 persist.vendor.face.virtual.type u:object_r:virtual_face_prop:s0 exact string
