--- a/vendor-skels/google_devices/husky/grapheneos/Android.bp
+++ b/vendor-skels/google_devices/husky/grapheneos/Android.bp
@@ -0,0 +1,6 @@
+prebuilt_etc {
+    name: "init.husky.grapheneos.rc",
+    src: "init.husky.grapheneos.rc",
+    sub_dir: "init",
+    vendor: true,
+}
--- a/vendor-skels/google_devices/husky/grapheneos/init.husky.grapheneos.rc
+++ b/vendor-skels/google_devices/husky/grapheneos/init.husky.grapheneos.rc
@@ -0,0 +1,47 @@
+on property:persist.sys.husky_high_emission_frequency_mode=0x00 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x00,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x01 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x01,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x02 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x02,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x03 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x03,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x04 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x04,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x05 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x05,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x06 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x06,0x43,0x43,0x03
+
+on property:persist.sys.husky_high_emission_frequency_mode=0x07 && property:sys.state_of_default_display=on
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x07,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x00
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x00,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x01
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x01,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x02
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x02,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x03
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x03,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x04
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x04,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x05
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x05,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x06
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x06,0x43,0x43,0x03
+
+on property:sys.state_of_default_display=on && property:persist.sys.husky_high_emission_frequency_mode=0x07
+    write /sys/module/panel_google_hk3/parameters/freq_cmd 0x07,0x43,0x43,0x03
--- a/vendor-skels/google_devices/husky/husky.mk
+++ b/vendor-skels/google_devices/husky/husky.mk
@@ -12,7 +12,10 @@ PRODUCT_SOONG_NAMESPACES += \
     vendor/google_devices/husky/overlays \
     vendor/google_devices/husky/proprietary \
     vendor/google_devices/husky/sysconfig \
-    vendor/google_devices/husky/vintf
+    vendor/google_devices/husky/vintf \
+    vendor/google_devices/husky/grapheneos
+
+PRODUCT_PACKAGES += init.husky.grapheneos.rc
 
 PRODUCT_NAME := husky
 PRODUCT_DEVICE := husky
