diff --git a/res/values/strings.xml b/res/values/strings.xml
index 1d76e61..ee89ae0 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -12090,6 +12090,10 @@
 
     <!-- Title for the screen to show all the 2- and 3-button navigation settings. [CHAR LIMIT=80] -->
     <string name="button_navigation_settings_activity_title">Button navigation</string>
+    <string name="show_taskbar_title">Show taskbar</string>
+    <string name="show_taskbar_summary">Use the tablet taskbar instead of phone-style navigation</string>
+    <string name="show_navbar_hint_title">Show navigation handle</string>
+    <string name="show_navbar_hint_summary">Keep the gesture navigation handle visible when the taskbar is hidden</string>
 
     <!-- Title text for changing the order of the navbar buttons -->
     <string name="button_navigation_settings_order_title">Button order</string>
diff --git a/res/xml/button_navigation_settings.xml b/res/xml/button_navigation_settings.xml
index 1f89539..b4bf82a 100644
--- a/res/xml/button_navigation_settings.xml
+++ b/res/xml/button_navigation_settings.xml
@@ -31,6 +31,12 @@
         settings:lottie_rawRes="@raw/lottie_button_nav_menu"
         settings:controller="com.android.settings.gestures.ButtonNavigationSettingsAssistController"/>
 
+    <SwitchPreferenceCompat
+        android:key="button_navigation_show_taskbar"
+        android:title="@string/show_taskbar_title"
+        android:summary="@string/show_taskbar_summary"
+        settings:controller="com.android.settings.gestures.TaskbarEnabledPreferenceController" />
+
     <SwitchPreferenceCompat
         android:key="assistant_long_press_home_gesture"
         android:title="@string/assistant_long_press_home_gesture_title"
@@ -39,4 +45,3 @@
     />
 
 </PreferenceScreen>
-
diff --git a/res/xml/gesture_navigation_settings.xml b/res/xml/gesture_navigation_settings.xml
index d271985..46f2d2f 100644
--- a/res/xml/gesture_navigation_settings.xml
+++ b/res/xml/gesture_navigation_settings.xml
@@ -34,6 +34,18 @@
             settings:buttonPreferenceType="filled" />
     </PreferenceCategory>
 
+    <SwitchPreferenceCompat
+        android:key="gesture_navigation_show_taskbar"
+        android:title="@string/show_taskbar_title"
+        android:summary="@string/show_taskbar_summary"
+        settings:controller="com.android.settings.gestures.TaskbarEnabledPreferenceController" />
+
+    <SwitchPreferenceCompat
+        android:key="gesture_navigation_show_navbar_hint"
+        android:title="@string/show_navbar_hint_title"
+        android:summary="@string/show_navbar_hint_summary"
+        settings:controller="com.android.settings.gestures.NavigationBarHintPreferenceController" />
+
     <PreferenceCategory
         android:key="assistant_gesture_category"
         android:persistent="false"
@@ -75,4 +87,3 @@
         android:title="@string/back_sensitivity_dialog_message"
         settings:searchable="false"/>
 </PreferenceScreen>
-
diff --git a/src/com/android/settings/gestures/NavigationBarHintPreferenceController.java b/src/com/android/settings/gestures/NavigationBarHintPreferenceController.java
new file mode 100644
index 0000000..464b1b8
--- /dev/null
+++ b/src/com/android/settings/gestures/NavigationBarHintPreferenceController.java
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settings.gestures;
+
+import android.content.Context;
+
+import com.android.settings.R;
+import com.android.settings.core.TogglePreferenceController;
+
+public class NavigationBarHintPreferenceController extends TogglePreferenceController {
+
+    public NavigationBarHintPreferenceController(Context context, String preferenceKey) {
+        super(context, preferenceKey);
+    }
+
+    @Override
+    public boolean isChecked() {
+        return TaskbarSettingsUtils.isChecked(
+                mContext, TaskbarSettingsUtils.SETTING_NAVIGATION_BAR_HINT);
+    }
+
+    @Override
+    public boolean setChecked(boolean isChecked) {
+        return TaskbarSettingsUtils.setChecked(
+                mContext, TaskbarSettingsUtils.SETTING_NAVIGATION_BAR_HINT, isChecked);
+    }
+
+    @Override
+    public int getAvailabilityStatus() {
+        return TaskbarSettingsUtils.isNavigationBarHintSettingAvailable(mContext)
+                ? AVAILABLE : UNSUPPORTED_ON_DEVICE;
+    }
+
+    @Override
+    public int getSliceHighlightMenuRes() {
+        return R.string.menu_key_system;
+    }
+}
diff --git a/src/com/android/settings/gestures/TaskbarEnabledPreferenceController.java b/src/com/android/settings/gestures/TaskbarEnabledPreferenceController.java
new file mode 100644
index 0000000..2e9bdf1
--- /dev/null
+++ b/src/com/android/settings/gestures/TaskbarEnabledPreferenceController.java
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settings.gestures;
+
+import android.content.Context;
+
+import com.android.settings.R;
+import com.android.settings.core.TogglePreferenceController;
+
+public class TaskbarEnabledPreferenceController extends TogglePreferenceController {
+
+    public TaskbarEnabledPreferenceController(Context context, String preferenceKey) {
+        super(context, preferenceKey);
+    }
+
+    @Override
+    public boolean isChecked() {
+        return TaskbarSettingsUtils.isChecked(mContext, TaskbarSettingsUtils.SETTING_ENABLE_TASKBAR);
+    }
+
+    @Override
+    public boolean setChecked(boolean isChecked) {
+        return TaskbarSettingsUtils.setChecked(
+                mContext, TaskbarSettingsUtils.SETTING_ENABLE_TASKBAR, isChecked);
+    }
+
+    @Override
+    public int getAvailabilityStatus() {
+        return TaskbarSettingsUtils.isTaskbarSettingAvailable(mContext)
+                ? AVAILABLE : UNSUPPORTED_ON_DEVICE;
+    }
+
+    @Override
+    public int getSliceHighlightMenuRes() {
+        return R.string.menu_key_system;
+    }
+}
diff --git a/src/com/android/settings/gestures/TaskbarSettingsUtils.java b/src/com/android/settings/gestures/TaskbarSettingsUtils.java
new file mode 100644
index 0000000..2bb88b0
--- /dev/null
+++ b/src/com/android/settings/gestures/TaskbarSettingsUtils.java
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settings.gestures;
+
+import android.content.Context;
+import android.provider.Settings;
+
+final class TaskbarSettingsUtils {
+    private static final int MIN_LARGE_SCREEN_WIDTH_DP = 600;
+
+    static final int ON = 1;
+    static final int OFF = 0;
+    static final String SETTING_ENABLE_TASKBAR = "enable_taskbar";
+    static final String SETTING_NAVIGATION_BAR_HINT = "navigation_bar_hint";
+
+    private TaskbarSettingsUtils() {
+    }
+
+    static boolean isTaskbarSettingAvailable(Context context) {
+        return context.getResources().getConfiguration().smallestScreenWidthDp
+                >= MIN_LARGE_SCREEN_WIDTH_DP;
+    }
+
+    static boolean isNavigationBarHintSettingAvailable(Context context) {
+        return true;
+    }
+
+    static boolean isChecked(Context context, String key) {
+        return Settings.System.getInt(context.getContentResolver(), key, ON) == ON;
+    }
+
+    static boolean setChecked(Context context, String key, boolean enabled) {
+        return Settings.System.putInt(context.getContentResolver(), key, enabled ? ON : OFF);
+    }
+}
