--- a/src/com/android/settings/development/AdbIpAddressPreferenceController.java	2026-06-25 19:02:36.962721210 +1000
+++ b/src/com/android/settings/development/AdbIpAddressPreferenceController.java	2026-06-25 19:03:06.254722021 +1000
@@ -121,11 +121,7 @@
     protected void updateConnectivity() {
         String ipAddress = getIpv4Address();
         int port = getPort();
-        if (!AdbWirelessDebuggingPreferenceController.isWifiConnected(mContext)) {
-            mAdbIpAddrPref.setSummary(mContext.getString(
-                    com.android.settingslib.R.string.adb_wireless_unavailable_disconnected
-            ));
-        } else if (port <= 0) {
+        if (port <= 0) {
             mAdbIpAddrPref.setSummary(mContext.getString(
                     com.android.settingslib.R.string.adb_wireless_unavailable_not_allowed
             ));
--- a/src/com/android/settings/development/AdbWirelessDebuggingEnabler.java	2026-06-25 19:02:36.962721210 +1000
+++ b/src/com/android/settings/development/AdbWirelessDebuggingEnabler.java	2026-06-25 19:03:06.254722021 +1000
@@ -24,7 +24,6 @@
 import android.os.Looper;
 import android.provider.Settings;
 import android.util.Log;
-import android.widget.Toast;
 
 import com.android.settings.widget.SwitchWidgetController;
 import com.android.settingslib.core.lifecycle.Lifecycle;
@@ -122,11 +121,6 @@
 
     @Override
     public boolean onSwitchToggled(boolean isChecked) {
-        if (isChecked && !AdbWirelessDebuggingPreferenceController.isWifiConnected(mContext)) {
-            Toast.makeText(mContext, com.android.settingslib.R.string.adb_wireless_no_network_msg,
-                            Toast.LENGTH_LONG)
-                    .show();
-        }
         writeAdbWifiSetting(isChecked);
         return true;
     }
--- a/src/com/android/settings/development/AdbWirelessDebuggingPreferenceController.java	2026-06-25 19:02:36.962721210 +1000
+++ b/src/com/android/settings/development/AdbWirelessDebuggingPreferenceController.java	2026-06-25 19:03:06.255722022 +1000
@@ -19,9 +19,6 @@
 import android.content.Context;
 import android.database.ContentObserver;
 import android.debug.IAdbManager;
-import android.net.ConnectivityManager;
-import android.net.Network;
-import android.net.NetworkCapabilities;
 import android.net.Uri;
 import android.os.Handler;
 import android.os.Looper;
@@ -29,7 +26,6 @@
 import android.os.ServiceManager;
 import android.provider.Settings;
 import android.util.Log;
-import android.widget.Toast;
 
 import androidx.preference.Preference;
 import androidx.preference.PreferenceScreen;
@@ -143,38 +139,9 @@
         ((PrimarySwitchPreference) preference).setChecked(enabled);
     }
 
-    /**
-     * Returns true if connected to Wi-Fi network.
-     */
-    public static boolean isWifiConnected(Context context) {
-        ConnectivityManager cm = context.getSystemService(ConnectivityManager.class);
-        if (cm == null) {
-            return false;
-        }
-        for (Network network : cm.getAllNetworks()) {
-            final NetworkCapabilities nc = cm.getNetworkCapabilities(network);
-            if (nc == null) {
-                continue;
-            }
-            if (nc.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
-                    // NET_CAPABILITY_INTERNET to filter out hotspot networks.
-                    && nc.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)) {
-                return true;
-            }
-        }
-        return false;
-    }
-
     @Override
     public boolean onPreferenceChange(Preference preference, Object newValue) {
         final boolean enabled = (Boolean) newValue;
-        if (enabled && !isWifiConnected(mContext)) {
-            Toast.makeText(
-                    mContext, com.android.settingslib.R.string.adb_wireless_no_network_msg,
-                            Toast.LENGTH_LONG)
-                    .show();
-        }
-
         Settings.Global.putInt(mContext.getContentResolver(),
                 Settings.Global.ADB_WIFI_ENABLED,
                 enabled ? AdbPreferenceController.ADB_SETTING_ON
--- a/src/com/android/settings/development/qstile/AdbWirelessDebuggingDevelopmentTile.java	2026-06-25 19:02:36.973721210 +1000
+++ b/src/com/android/settings/development/qstile/AdbWirelessDebuggingDevelopmentTile.java	2026-06-25 19:03:06.256722022 +1000
@@ -27,7 +27,6 @@
 import android.os.Handler;
 import android.os.Looper;
 import android.provider.Settings;
-import android.widget.Toast;
 
 import com.android.settings.development.AdbWirelessDebuggingPreferenceController;
 
@@ -37,7 +36,6 @@
 public class AdbWirelessDebuggingDevelopmentTile extends DevelopmentTiles {
     private Context mContext;
     private KeyguardManager mKeyguardManager;
-    private Toast mToast;
     private final Handler mHandler = new Handler(Looper.getMainLooper());
     private final ContentObserver mSettingsObserver = new ContentObserver(mHandler) {
         @Override
@@ -51,9 +49,6 @@
         super.onCreate();
         mContext = getApplicationContext();
         mKeyguardManager = mContext.getSystemService(KeyguardManager.class);
-        mToast = Toast.makeText(mContext,
-                com.android.settingslib.R.string.adb_wireless_no_network_msg,
-                Toast.LENGTH_LONG);
     }
 
     @Override
@@ -82,13 +77,6 @@
             return;
         }
 
-        // Show error toast if not connected to Wi-Fi
-        if (isEnabled && !AdbWirelessDebuggingPreferenceController.isWifiConnected(mContext)) {
-            // Close quick shade
-            sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
-            mToast.show();
-        }
-
         writeAdbWifiSetting(isEnabled);
     }
 
