--- a/subprojects/core/src/main/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactory.java
+++ b/subprojects/core/src/main/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactory.java
@@ -246 +246 @@
-            Annotation optional = annotationTarget.getAnnotation(Optional.class);
+            Annotation optional = annotationTarget.getAnnotation(org.gradle.api.tasks.Optional.class);
--- a/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/mvnsettings/DefaultLocalMavenRepositoryLocator.java
+++ b/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/mvnsettings/DefaultLocalMavenRepositoryLocator.java
@@ -18,2 +18,2 @@
-import jarjar.org.apache.maven.settings.Settings;
-import jarjar.org.apache.maven.settings.building.*;
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.building.*;
--- a/subprojects/plugins/src/main/groovy/org/gradle/api/tasks/javadoc/Groovydoc.java
+++ b/subprojects/plugins/src/main/groovy/org/gradle/api/tasks/javadoc/Groovydoc.java
@@ -162 +162 @@
-    @Input @Optional
+    @Input @org.gradle.api.tasks.Optional
@@ -179 +179 @@
-    @Input @Optional
+    @Input @org.gradle.api.tasks.Optional
@@ -196 +196 @@
-    @Input @Optional
+    @Input @org.gradle.api.tasks.Optional
@@ -213 +213 @@
-    @Input @Optional
+    @Input @org.gradle.api.tasks.Optional
