--- a/BaseBin/dyldhook/src/main.S
+++ b/BaseBin/dyldhook/src/main.S
@@ -107,4 +107,39 @@
 .data
 .global ___stack_chk_guard
 ___stack_chk_guard:
-    .quad 0x1337
\ No newline at end of file
+    .quad 0x1337
+// WebKitPlayground / DYLD_FRAMEWORK_PATH over DSC (Lessica)
+.global _MACHOMERGER_ORIG__ZNK5dyld413ProcessConfig9DyldCache17isOverridablePathEPKc
+.global _MACHOMERGER_HOOKTRAMPOLINE__ZNK5dyld413ProcessConfig9DyldCache17isOverridablePathEPKc
+_MACHOMERGER_ORIG__ZNK5dyld413ProcessConfig9DyldCache17isOverridablePathEPKc:
+    nop
+    // FALLTHROUGH
+_MACHOMERGER_HOOKTRAMPOLINE__ZNK5dyld413ProcessConfig9DyldCache17isOverridablePathEPKc:
+    b .
+
+.global _MACHOMERGER_ORIG__ZN5dyld413ProcessConfig9DyldCache23isAlwaysOverridablePathEPKc
+.global _MACHOMERGER_HOOKTRAMPOLINE__ZN5dyld413ProcessConfig9DyldCache23isAlwaysOverridablePathEPKc
+_MACHOMERGER_ORIG__ZN5dyld413ProcessConfig9DyldCache23isAlwaysOverridablePathEPKc:
+    nop
+    // FALLTHROUGH
+_MACHOMERGER_HOOKTRAMPOLINE__ZN5dyld413ProcessConfig9DyldCache23isAlwaysOverridablePathEPKc:
+    b .
+
+.global _MACHOMERGER_ORIG__ZNK5dyld46Loader17applyCachePatchesERNS_12RuntimeStateERNS_34DyldCacheDataConstLazyScopedWriterE
+.global _MACHOMERGER_HOOKTRAMPOLINE__ZNK5dyld46Loader17applyCachePatchesERNS_12RuntimeStateERNS_34DyldCacheDataConstLazyScopedWriterE
+_MACHOMERGER_ORIG__ZNK5dyld46Loader17applyCachePatchesERNS_12RuntimeStateERNS_34DyldCacheDataConstLazyScopedWriterE:
+    nop
+    // FALLTHROUGH
+_MACHOMERGER_HOOKTRAMPOLINE__ZNK5dyld46Loader17applyCachePatchesERNS_12RuntimeStateERNS_34DyldCacheDataConstLazyScopedWriterE:
+    b .
+
+MAKE_TRAMPOLINE(_ZNK5dyld46Loader21overridesDylibInCacheERPKNS0_10DylibPatchERt)
+MAKE_TRAMPOLINE(_ZNK5dyld46Loader11loadAddressERNS_12RuntimeStateE)
+
+.global _MACHOMERGER_ORIG__ZNK5dyld416JustInTimeLoader11matchesPathEPKc
+.global _MACHOMERGER_HOOKTRAMPOLINE__ZNK5dyld416JustInTimeLoader11matchesPathEPKc
+_MACHOMERGER_ORIG__ZNK5dyld416JustInTimeLoader11matchesPathEPKc:
+    nop
+    // FALLTHROUGH
+_MACHOMERGER_HOOKTRAMPOLINE__ZNK5dyld416JustInTimeLoader11matchesPathEPKc:
+    b .
