diff --git a/README.md b/README.md
index 4aa3832e..2347cd63 100644
--- a/README.md
+++ b/README.md
@@ -11,6 +11,10 @@
 
 Connecting things to your VPN made simple. Share your VPN connection over hotspot or repeater. (**root required**)  
 
+[<img src="https://fdroid.gitlab.io/artwork/badge/get-it-on.png"
+     alt="Get it on F-Droid"
+     height="80">](https://f-droid.org/packages/be.mygod.vpnhotspot.foss/)
+
 This app is useful for:
 
 * Connecting things that don't support VPN like Chromecasts behind corporate firewalls;
diff --git a/build.gradle.kts b/build.gradle.kts
index 33e92070..a3e1808e 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,8 +1,8 @@
 plugins {
     alias(libs.plugins.android.application) apply false
     alias(libs.plugins.ben.manes.versions)
-    alias(libs.plugins.crashlytics) apply false
-    alias(libs.plugins.google.services) apply false
+    //alias(libs.plugins.crashlytics) apply false
+    //alias(libs.plugins.google.services) apply false
     alias(libs.plugins.kotlin.android) apply false
     alias(libs.plugins.ksp) apply false
 }
diff --git a/fastlane/metadata/android/en-US/changelogs/257.txt b/fastlane/metadata/android/en-US/changelogs/257.txt
new file mode 100644
index 00000000..45f91068
--- /dev/null
+++ b/fastlane/metadata/android/en-US/changelogs/257.txt
@@ -0,0 +1 @@
+Fixes.
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/changelogs/259.txt b/fastlane/metadata/android/en-US/changelogs/259.txt
new file mode 100644
index 00000000..e321e75b
--- /dev/null
+++ b/fastlane/metadata/android/en-US/changelogs/259.txt
@@ -0,0 +1,2 @@
+* Refined MAC address resolving.
+* Fixes.
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/changelogs/260.txt b/fastlane/metadata/android/en-US/changelogs/260.txt
new file mode 100644
index 00000000..2ed9b969
--- /dev/null
+++ b/fastlane/metadata/android/en-US/changelogs/260.txt
@@ -0,0 +1,3 @@
+* Initial support for Android 12 beta 1. (Support for new features will land later)
+* Support turning off hardware tether acceleration on Android 8- if the ROM supports it. (#224)
+* Fixes.
diff --git a/fastlane/metadata/android/en-US/full_description.txt b/fastlane/metadata/android/en-US/full_description.txt
new file mode 100644
index 00000000..34aafd5c
--- /dev/null
+++ b/fastlane/metadata/android/en-US/full_description.txt
@@ -0,0 +1,15 @@
+Connecting things to your VPN made simple. Share your VPN connection over hotspot/system tethering or repeater. (root required)
+
+This app is useful for:
+
+* Connecting things that don't support VPN like Chromecasts behind corporate firewalls;
+* Setting up gapps behind corporate firewalls;
+* Connecting to your mobile hotspot but you're not bothered to set up VPN on your device;
+* Identifying, monitoring and blocking (unwanted) clients;
+* Bypassing tethering limits by:
+- (recommended) Use this app with a real VPN/socksifier; (ad-blocker may not work)
+- (might or might not work based on your carrier) Use this app with TTL editor like
+TTL Master;
+- Try your luck and simply use this app.
+
+Get more information including FAQs on GitHub: https://github.com/Mygod/VPNHotspot/blob/master/README.md
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/1.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/1.png
new file mode 100644
index 00000000..c87c7188
Binary files /dev/null and b/fastlane/metadata/android/en-US/images/phoneScreenshots/1.png differ
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/2.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/2.png
new file mode 100644
index 00000000..a65d4fc8
Binary files /dev/null and b/fastlane/metadata/android/en-US/images/phoneScreenshots/2.png differ
diff --git a/fastlane/metadata/android/en-US/short_description.txt b/fastlane/metadata/android/en-US/short_description.txt
new file mode 100644
index 00000000..454eeb23
--- /dev/null
+++ b/fastlane/metadata/android/en-US/short_description.txt
@@ -0,0 +1 @@
+tethering/Wi-Fi repeater
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/title.txt b/fastlane/metadata/android/en-US/title.txt
new file mode 100644
index 00000000..e91b71e4
--- /dev/null
+++ b/fastlane/metadata/android/en-US/title.txt
@@ -0,0 +1 @@
+VPN Hotspot
\ No newline at end of file
diff --git a/mobile/build.gradle.kts b/mobile/build.gradle.kts
index aec241bc..0112729c 100644
--- a/mobile/build.gradle.kts
+++ b/mobile/build.gradle.kts
@@ -1,12 +1,12 @@
 import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 
 plugins {
-    alias(libs.plugins.aboutLibraries)
+    //alias(libs.plugins.aboutLibraries)
     alias(libs.plugins.android.application)
-    alias(libs.plugins.crashlytics)
+    //alias(libs.plugins.crashlytics)
     alias(libs.plugins.kotlin.compose)
     alias(libs.plugins.ksp)
-    alias(libs.plugins.google.services)
+    //alias(libs.plugins.google.services)
     kotlin("android")
     kotlin("kapt")
     id("kotlin-parcelize")
@@ -58,6 +58,14 @@ android {
     ))
     lint.warning += "FullBackupContent"
     lint.warning += "UnsafeOptInUsageError"
+    flavorDimensions.add("freedom")
+    productFlavors {
+        create("freedom") {
+            applicationId = "be.mygod.vpnhotspot_foss"
+            dimension = "freedom"
+            //resourceConfigurations.addAll(arrayOf("es", "it", "pt-rBR", "ru", "zh-rCN", "zh-rTW"))
+        }
+    }
     sourceSets.getByName("androidTest").assets.srcDir("$projectDir/schemas")
     externalNativeBuild.cmake.path = file("src/main/cpp/CMakeLists.txt")
 }
@@ -71,15 +79,15 @@ kotlin.compilerOptions.jvmTarget.set(JvmTarget.fromTarget(javaVersion.toString()
 dependencies {
     coreLibraryDesugaring(libs.desugar.jdk.libs)
     ksp(libs.room.compiler)
-    implementation(libs.aboutlibraries.compose.m3)
+    //implementation(libs.aboutlibraries.compose.m3)
     implementation(libs.activity.compose)
     implementation(libs.browser)
     implementation(libs.core.i18n)
     implementation(libs.core.ktx)
     implementation(libs.dexmaker)
     implementation(libs.dnsjava)
-    implementation(libs.firebase.analytics)
-    implementation(libs.firebase.crashlytics)
+    //implementation(libs.firebase.analytics)
+    //implementation(libs.firebase.crashlytics)
     implementation(libs.foundation.layout)
     implementation(libs.fragment.ktx)
     implementation(libs.hiddenapibypass)
@@ -91,7 +99,7 @@ dependencies {
     implementation(libs.lifecycle.runtime.ktx)
     implementation(libs.material)
     implementation(libs.material3.android)
-    implementation(libs.play.services.oss.licenses)
+    //implementation(libs.play.services.oss.licenses)
     implementation(libs.preference)
     implementation(libs.preferencex.simplemenu)
     implementation(libs.room.ktx)
diff --git a/mobile/src/main/AndroidManifest.xml b/mobile/src/main/AndroidManifest.xml
index 1172d87d..0ead4f7b 100644
--- a/mobile/src/main/AndroidManifest.xml
+++ b/mobile/src/main/AndroidManifest.xml
@@ -267,7 +267,7 @@
 
         <provider
             android:name="androidx.core.content.FileProvider"
-            android:authorities="be.mygod.vpnhotspot.log"
+            android:authorities="be.mygod.vpnhotspot.foss.log"
             android:exported="false"
             android:grantUriPermissions="true">
             <meta-data
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/AboutLibrariesActivity.kt b/mobile/src/main/java/be/mygod/vpnhotspot/AboutLibrariesActivity.kt
index 87be3508..779de04c 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/AboutLibrariesActivity.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/AboutLibrariesActivity.kt
@@ -22,15 +22,11 @@ import androidx.compose.ui.platform.UriHandler
 import androidx.compose.ui.res.colorResource
 import androidx.compose.ui.res.stringResource
 import be.mygod.vpnhotspot.util.launchUrl
-import com.mikepenz.aboutlibraries.Libs
-import com.mikepenz.aboutlibraries.ui.compose.m3.LibrariesContainer
-import com.mikepenz.aboutlibraries.util.withContext
-
 class AboutLibrariesActivity : ComponentActivity(), UriHandler {
     @OptIn(ExperimentalMaterial3Api::class)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        enableEdgeToEdge()
+        /*enableEdgeToEdge()
         setContent {
             MaterialTheme(
                 if (isSystemInDarkTheme()) darkColorScheme(
@@ -60,7 +56,7 @@ class AboutLibrariesActivity : ComponentActivity(), UriHandler {
                     }
                 }
             }
-        }
+        }*/
     }
 
     override fun openUri(uri: String) = launchUrl(uri)
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/App.kt b/mobile/src/main/java/be/mygod/vpnhotspot/App.kt
index 5cbd9364..2fafb618 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/App.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/App.kt
@@ -29,13 +29,6 @@ import be.mygod.vpnhotspot.util.DeviceStorageApp
 import be.mygod.vpnhotspot.util.Services
 import be.mygod.vpnhotspot.util.privateLookup
 import be.mygod.vpnhotspot.widget.SmartSnackbar
-import com.google.android.gms.dynamite.DynamiteModule
-import com.google.firebase.analytics.ParametersBuilder
-import com.google.firebase.analytics.ktx.analytics
-import com.google.firebase.analytics.logEvent
-import com.google.firebase.crashlytics.FirebaseCrashlytics
-import com.google.firebase.ktx.Firebase
-import com.google.firebase.provider.FirebaseInitProvider
 import kotlinx.coroutines.DEBUG_PROPERTY_NAME
 import kotlinx.coroutines.DEBUG_PROPERTY_VALUE_ON
 import kotlinx.coroutines.GlobalScope
@@ -63,31 +56,23 @@ class App : Application() {
 
         // overhead of debug mode is minimal: https://github.com/Kotlin/kotlinx.coroutines/blob/f528898/docs/debugging.md#debug-mode
         System.setProperty(DEBUG_PROPERTY_NAME, DEBUG_PROPERTY_VALUE_ON)
-        DynamiteModule::class.java.getDeclaredField("zzf").apply { isAccessible = true }.set(null, false)
+        //DynamiteModule::class.java.getDeclaredField("zzf").apply { isAccessible = true }.set(null, false)
         // call super.attachInfo get around ProviderInfo check
-        FirebaseInitProvider::class.java.privateLookup().findSpecial(ContentProvider::class.java, "attachInfo",
-            MethodType.methodType(Void.TYPE, Context::class.java, ProviderInfo::class.java),
-            FirebaseInitProvider::class.java).bindTo(FirebaseInitProvider()).invokeWithArguments(deviceStorage, null)
-        FirebaseCrashlytics.getInstance().apply {
-            setCustomKey("uname.release", Os.uname().release)
-            setCustomKey("build", Build.DISPLAY)
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) setCustomKey("extension_s",
-                SdkExtensions.getExtensionVersion(Build.VERSION_CODES.S))
-        }
+
         Timber.plant(object : Timber.DebugTree() {
             @SuppressLint("LogNotTimber")
             override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
                 if (t == null) {
                     if (priority != Log.DEBUG || BuildConfig.DEBUG) Log.println(priority, tag, message)
-                    FirebaseCrashlytics.getInstance().log("${"XXVDIWEF".getOrElse(priority) { 'X' }}/$tag: $message")
+                    //FirebaseCrashlytics.getInstance().log("${"XXVDIWEF".getOrElse(priority) { 'X' }}/$tag: $message")
                 } else {
                     if (priority >= Log.WARN || priority == Log.DEBUG) {
                         Log.println(priority, tag, message)
                         Log.w(tag, message, t)
                     }
-                    if (priority >= Log.INFO && t !is NoShellException) {
-                        FirebaseCrashlytics.getInstance().recordException(t)
-                    }
+                    //if (priority >= Log.INFO && t !is NoShellException) {
+                        //FirebaseCrashlytics.getInstance().recordException(t)
+                    //}
                 }
             }
         })
@@ -111,11 +96,14 @@ class App : Application() {
      * This method is used to log "expected" and well-handled errors, i.e. we care less about logs, etc.
      * logException is inappropriate sometimes because it flushes all logs that could be used to investigate other bugs.
      */
-    fun logEvent(@Size(min = 1L, max = 40L) event: String, block: ParametersBuilder.() -> Unit = { }) {
+    /*fun logEvent(@Size(min = 1L, max = 40L) event: String, block: ParametersBuilder.() -> Unit = { }) {
         Firebase.analytics.logEvent(event) {
             block(this)
             Timber.i(if (bundle.isEmpty) event else "$event, extras: $bundle")
         }
+    }*/
+    fun logEvent(@Size(min = 1L, max = 40L) event: String, t : Throwable) {
+        Timber.i(if (t == null) event else "$event, extras: $t")
     }
 
     /**
@@ -128,7 +116,7 @@ class App : Application() {
             context.startActivity(Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS))
             Toast.makeText(context, R.string.tethering_location_off, Toast.LENGTH_LONG).show()
         } catch (e: ActivityNotFoundException) {
-            app.logEvent("location_settings") { param("message", e.toString()) }
+            app.logEvent("location_settings", e) //{ param("message", e.toString()) }
             SmartSnackbar.make(R.string.tethering_location_off).show()
         } else context.startForegroundService(Intent(context, T::class.java))
     }
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/SettingsPreferenceFragment.kt b/mobile/src/main/java/be/mygod/vpnhotspot/SettingsPreferenceFragment.kt
index de48f917..b271a97d 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/SettingsPreferenceFragment.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/SettingsPreferenceFragment.kt
@@ -138,10 +138,10 @@ class SettingsPreferenceFragment : PreferenceFragmentCompat() {
             requireContext().launchUrl("https://mygod.be/donate/")
             true
         }
-        findPreference<Preference>("misc.licenses")!!.setOnPreferenceClickListener {
-            startActivity(Intent(requireContext(), AboutLibrariesActivity::class.java))
+        /*findPreference<Preference>("misc.licenses")!!.setOnPreferenceClickListener {
+            startActivity(Intent(context, OssLicensesMenuActivity::class.java))
             true
-        }
+        }*/
     }
 
     override fun onDisplayPreferenceDialog(preference: Preference) = when (preference.key) {
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/manage/ManageBar.kt b/mobile/src/main/java/be/mygod/vpnhotspot/manage/ManageBar.kt
index 4ec1b51f..83d4055d 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/manage/ManageBar.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/manage/ManageBar.kt
@@ -35,12 +35,12 @@ object ManageBar : Manager() {
         } catch (e1: RuntimeException) {
             try {
                 startActivity(intent.setClassName(SETTINGS_PACKAGE, SETTINGS_2))
-                app.logEvent(TAG) { param(SETTINGS_1, e1.toString()) }
+                app.logEvent(TAG, e1) //{ param(SETTINGS_1, e1.toString()) }
             } catch (e2: RuntimeException) {
-                app.logEvent(TAG) {
-                    param(SETTINGS_1, e1.toString())
-                    param(SETTINGS_2, e2.toString())
-                }
+                app.logEvent(TAG, e2) /*{
+                     param(SETTINGS_1, e1.toString())
+                     param(SETTINGS_2, e2.toString())
+                }*/
             }
         }
     }
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/manage/TetherManager.kt b/mobile/src/main/java/be/mygod/vpnhotspot/manage/TetherManager.kt
index 08823300..7bde0c13 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/manage/TetherManager.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/manage/TetherManager.kt
@@ -62,7 +62,6 @@ sealed class TetherManager(protected val parent: TetheringFragment) : Manager(),
                         "package:${mainActivity.packageName}".toUri()))
                 return
             } catch (e: RuntimeException) {
-                app.logEvent("manage_write_settings") { param("message", e.toString()) }
             }
             when (manager.isStarted) {
                 true -> try {
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/net/monitor/IpMonitor.kt b/mobile/src/main/java/be/mygod/vpnhotspot/net/monitor/IpMonitor.kt
index aafa1392..c5eb7878 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/net/monitor/IpMonitor.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/net/monitor/IpMonitor.kt
@@ -121,7 +121,6 @@ abstract class IpMonitor {
                     Timber.w(e)
                 }
                 if (destroyed) return@launch
-                app.logEvent("ip_monitor_failure")
             }
             withContext(Dispatchers.IO) {
                 var server: RootServer? = null
@@ -155,7 +154,6 @@ abstract class IpMonitor {
             poll()
             return server
         } catch (e: IOException) {
-            app.logEvent("ip_poll_failure")
             Timber.d(e)
         }
         var newServer = server
@@ -168,7 +166,6 @@ abstract class IpMonitor {
             if (lines.any { errorMatcher.containsMatchIn(it) }) throw IOException(result.out)
             processLines(lines.asSequence())
         } catch (e: Exception) {
-            app.logEvent("ip_su_poll_failure") { param("cause", e.message.toString()) }
             Timber.d(e)
         }
         return if (newServer?.active != false) newServer else {
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/P2pSupplicantConfiguration.kt b/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/P2pSupplicantConfiguration.kt
index 286fb93d..46a0a4f4 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/P2pSupplicantConfiguration.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/P2pSupplicantConfiguration.kt
@@ -6,8 +6,6 @@ import be.mygod.vpnhotspot.RepeaterService
 import be.mygod.vpnhotspot.net.MacAddressCompat
 import be.mygod.vpnhotspot.root.RepeaterCommands
 import be.mygod.vpnhotspot.root.RootManager
-import com.google.firebase.crashlytics.CustomKeysAndValues
-import com.google.firebase.crashlytics.FirebaseCrashlytics
 
 /**
  * This parser is based on:
@@ -134,12 +132,7 @@ class P2pSupplicantConfiguration(private val group: WifiP2pGroup? = null) {
             }
             content = Content(result, target!!, persistentMacLine, legacy)
         } catch (e: Exception) {
-            FirebaseCrashlytics.getInstance().recordException(e, CustomKeysAndValues.Builder().apply {
-                putString("config", config)
-                ownerAddress?.let { putString("ownerAddress", it) }
-                putString("p2pGroup", group.toString())
-            }.build())
-            throw LoggedException(e)
+            throw e
         }
     }
     val psk by lazy { group?.passphrase ?: content.target.psk!! }
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/WifiP2pManagerHelper.kt b/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/WifiP2pManagerHelper.kt
index 4baa0ad5..1821fb4a 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/WifiP2pManagerHelper.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/net/wifi/WifiP2pManagerHelper.kt
@@ -50,7 +50,6 @@ object WifiP2pManagerHelper {
         try {
             setWifiP2pChannels(this, c, lc, oc, result)
         } catch (_: NoSuchMethodException) {
-            app.logEvent("NoSuchMethod_setWifiP2pChannels")
             return UNSUPPORTED
         }
         return result.future.await()
@@ -76,7 +75,6 @@ object WifiP2pManagerHelper {
             WifiP2pManager::class.java.getDeclaredMethod("startWps",
                     WifiP2pManager.Channel::class.java, WpsInfo::class.java, WifiP2pManager.ActionListener::class.java)
         } catch (_: NoSuchMethodException) {
-            app.logEvent("NoSuchMethod_startWps")
             null
         }
     }
@@ -101,7 +99,6 @@ object WifiP2pManagerHelper {
         try {
             deletePersistentGroup(this, c, netId, result)
         } catch (_: NoSuchMethodException) {
-            app.logEvent("NoSuchMethod_deletePersistentGroup")
             return UNSUPPORTED
         }
         return result.future.await()
diff --git a/mobile/src/main/java/be/mygod/vpnhotspot/root/RootManager.kt b/mobile/src/main/java/be/mygod/vpnhotspot/root/RootManager.kt
index 8de23eb0..fa84d4e1 100644
--- a/mobile/src/main/java/be/mygod/vpnhotspot/root/RootManager.kt
+++ b/mobile/src/main/java/be/mygod/vpnhotspot/root/RootManager.kt
@@ -14,7 +14,6 @@ import be.mygod.librootkotlinx.systemContext
 import be.mygod.vpnhotspot.App.Companion.app
 import be.mygod.vpnhotspot.util.Services
 import be.mygod.vpnhotspot.util.UnblockCentral
-import com.google.firebase.crashlytics.FirebaseCrashlytics
 import kotlinx.coroutines.CancellationException
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.GlobalScope
@@ -80,7 +79,7 @@ object RootManager : RootSession(), Logger {
     override suspend fun initServer(server: RootServer) {
         Logger.me = this
         AppProcess.shouldRelocateHeuristics.let {
-            FirebaseCrashlytics.getInstance().setCustomKey("RootManager.relocateEnabled", it)
+            //FirebaseCrashlytics.getInstance().setCustomKey("RootManager.relocateEnabled", it)
             server.init(app, it)
         }
         server.execute(RootInit())
diff --git a/mobile/src/main/res/xml/pref_settings.xml b/mobile/src/main/res/xml/pref_settings.xml
index 2f796fbc..dbbfed98 100644
--- a/mobile/src/main/res/xml/pref_settings.xml
+++ b/mobile/src/main/res/xml/pref_settings.xml
@@ -99,10 +99,10 @@
             app:icon="@drawable/ic_action_card_giftcard"
             app:title="@string/settings_misc_donate"
             app:summary="@string/settings_misc_donate_summary"/>
-        <Preference
+        <!--<Preference
             app:key="misc.licenses"
             app:icon="@drawable/ic_action_code"
             app:title="@string/oss_license_title"
-            app:summary="@string/preferences_license_summary"/>
+            app:summary="@string/preferences_license_summary"/>-->
     </PreferenceCategory>
 </PreferenceScreen>
