--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -23,7 +23,6 @@
     alias(libs.plugins.datadog) apply false
     alias(libs.plugins.devtools.ksp) apply false
     alias(libs.plugins.koin.compiler) apply false
-    alias(libs.plugins.firebase.crashlytics) apply false
     alias(libs.plugins.google.services) apply false
     alias(libs.plugins.room) apply false
     alias(libs.plugins.kotlin.android) apply false
