diff --git a/settings.gradle.kts b/settings.gradle.kts
index f32ada4..20362f3 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -33,7 +33,6 @@ pluginManagement {
 
 plugins {
     id("com.gradle.develocity") version "4.5.0"
-    id("org.gradle.toolchains.foojay-resolver") version "1.0.0"
     id("org.meshtastic.flatpak.sources.settings") version "0.1.5"
 }
 
@@ -87,16 +86,6 @@ develocity {
     }
 }
 
-@Suppress("UnstableApiUsage")
-toolchainManagement {
-    jvm {
-        javaRepositories {
-            repository("foojay") {
-                resolverClass.set(org.gradle.toolchains.foojay.FoojayToolchainResolver::class.java)
-            }
-        }
-    }
-}
 
 include(
     ":core:ble",
