--- a/buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/versioning/BuildVersionPlugin.kt
+++ b/buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/versioning/BuildVersionPlugin.kt
@@ -153,7 +153,7 @@
  */
 private
 fun Project.trimmedContentsOfFile(path: String): String =
-    fileContentsOf(path).asText.forUseAtConfigurationTime().get().trim()
+    fileContentsOf(path).asText.get().trim()
 
 
 private
@@ -179,7 +179,7 @@
                 isRunningInstallTask()
             })
         }
-    }.forUseAtConfigurationTime()
+    }
 
 
 private
@@ -196,10 +196,9 @@
                     .file(BuildReceipt.BUILD_RECEIPT_FILE_NAME)
                     .let(providers::fileContents)
                     .asText
-                    .forUseAtConfigurationTime()
             )
         }
-    }.forUseAtConfigurationTime()
+    }
 
 
 abstract class BuildTimestampValueSource : ValueSource<String, BuildTimestampValueSource.Parameters>, Describable {
@@ -298,12 +297,12 @@
 
 private
 fun Project.environmentVariable(variableName: String): Provider<String> =
-    providers.environmentVariable(variableName).forUseAtConfigurationTime()
+    providers.environmentVariable(variableName)
 
 
 private
 fun Project.gradleProperty(propertyName: String): Provider<String> =
-    providers.gradleProperty(propertyName).forUseAtConfigurationTime()
+    providers.gradleProperty(propertyName)
 
 
 /**
