diff --git a/app-thunderbird/build.gradle.kts b/app-thunderbird/build.gradle.kts
index af5e193..01dd868 100644
--- a/app-thunderbird/build.gradle.kts
+++ b/app-thunderbird/build.gradle.kts
@@ -256,10 +256,10 @@ dependencies {
 
     "fossImplementation"(projects.feature.funding.link)
 
-    fullDebugImplementation(projects.feature.funding.googleplay)
-    fullDailyImplementation(projects.feature.funding.googleplay)
-    fullBetaImplementation(projects.feature.funding.googleplay)
-    fullReleaseImplementation(projects.feature.funding.googleplay)
+    fullDebugImplementation(projects.feature.funding.link)
+    fullDailyImplementation(projects.feature.funding.link)
+    fullBetaImplementation(projects.feature.funding.link)
+    fullReleaseImplementation(projects.feature.funding.link)
 
     implementation(projects.feature.onboarding.migration.thunderbird)
     implementation(projects.feature.migration.launcher.thunderbird)
