--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -18,7 +18,6 @@
 }
 
 plugins {
-    id("org.gradle.toolchains.foojay-resolver-convention") version "1.0.0"
 }
 
 dependencyResolutionManagement {
@@ -30,12 +29,6 @@
                 includeGroupAndSubgroups("com.google")
             }
         }
-        maven(url = "https://maven.mozilla.org/maven2") {
-            mavenContent {
-                includeGroup("org.mozilla.components")
-                includeGroup("org.mozilla.telemetry")
-            }
-        }
         maven(url = "https://jitpack.io") {
             mavenContent {
                 includeGroup("com.github.ByteHamster")
@@ -41,7 +34,6 @@
 }
 
 include(
-    ":app-k9mail",
     ":app-thunderbird",
     ":app-ui-catalog",
 )
@@ -82,7 +74,6 @@
 
 include(
     ":feature:funding:api",
-    ":feature:funding:googleplay",
     ":feature:funding:link",
     ":feature:funding:noop",
 )
@@ -137,7 +128,6 @@
 include(
     ":feature:telemetry:api",
     ":feature:telemetry:noop",
-    ":feature:telemetry:glean",
 )
 
 include(
