diff --git a/settings.gradle.kts b/settings.gradle.kts
index 86e7950..84f2d29 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -18,7 +18,6 @@ pluginManagement {
 }
 
 plugins {
-    id("org.gradle.toolchains.foojay-resolver-convention") version "1.0.0"
 }
 
 dependencyResolutionManagement {
@@ -31,12 +30,6 @@ dependencyResolutionManagement {
                 includeGroupAndSubgroups("com.google")
             }
         }
-        maven(url = "https://maven.mozilla.org/maven2") {
-            mavenContent {
-                includeGroup("org.mozilla.components")
-                includeGroup("org.mozilla.telemetry")
-            }
-        }
         maven(url = "https://jitpack.io") {
             mavenContent {
                 includeGroup("com.github.ByteHamster")
@@ -70,7 +63,6 @@ if (useLocalComponents || useLocalBolt) {
 }
 
 include(
-    ":app-k9mail",
     ":app-thunderbird",
 )
 
@@ -111,7 +103,6 @@ include(
 
 include(
     ":feature:funding:api",
-    ":feature:funding:googleplay",
     ":feature:funding:link",
     ":feature:funding:noop",
 )
@@ -168,7 +159,6 @@ include(
 include(
     ":feature:telemetry:api",
     ":feature:telemetry:noop",
-    ":feature:telemetry:glean",
 )
 
 include(
