--- a/TMessagesProj/build.gradle
+++ b/TMessagesProj/build.gradle
@@ -71,7 +71,6 @@
     buildToolsVersion '35.0.0'
     ndkVersion "27.3.13750724" // 26.3.11579264
 
-    sourceSets.main.jniLibs.srcDirs = ['./jni/']
 
     externalNativeBuild {
         cmake {
--- a/TMessagesProj_App/build.gradle
+++ b/TMessagesProj_App/build.gradle
@@ -31,7 +31,6 @@
     buildToolsVersion '35.0.0'
 
     defaultConfig.applicationId = "org.forkclient.messenger"
 
-    sourceSets.main.jniLibs.srcDirs = ['../TMessagesProj/jni/']
 
     lintOptions {
