diff -ruN a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
--- a/buildSrc/build.gradle.kts	2026-05-01 03:47:57.294075299 +1000
+++ b/buildSrc/build.gradle.kts	2026-05-01 03:47:57.298075266 +1000
@@ -46,26 +46,16 @@
             implementation(gradleApi())
         }

-        afterEvaluate {
-            if (tasks.withType<ValidatePlugins>().isEmpty()) {
-                val validatePlugins by tasks.registering(ValidatePlugins::class) {
-                    outputFile.set(project.reporting.baseDirectory.file("task-properties/report.txt"))

-                    val mainSourceSet = project.sourceSets.main.get()
-                    classes.setFrom(mainSourceSet.output.classesDirs)
-                    dependsOn(mainSourceSet.output)
-                    classpath.setFrom(mainSourceSet.runtimeClasspath)
-                }
-                tasks.check { dependsOn(validatePlugins) }
-            }
-        }
+        apply(from = "../../../gradle/shared-with-buildSrc/code-quality-configuration.gradle.kts")
+    }
+}

-        tasks.withType<ValidatePlugins>().configureEach {
-            failOnWarning.set(true)
-            enableStricterValidation.set(true)
+subprojects {
+    tasks.all {
+        if (name.startsWith("compileTest") || name.startsWith("codenarc") || name.startsWith("checkstyle") || name == "test" || name == "check" || name.endsWith("Check") || name == "validateTaskProperties") {
+            enabled = false
         }
-
-        apply(from = "../../../gradle/shared-with-buildSrc/code-quality-configuration.gradle.kts")
     }
 }

diff -ruN a/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts b/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts
--- a/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts	2026-05-01 03:47:57.296075282 +1000
+++ b/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts	2026-05-01 03:47:57.302075234 +1000
@@ -135,7 +135,6 @@
 }

 fun Task.failedTaskCustomReports() = when (this) {
-    is ValidatePlugins -> listOf(outputFile.get().asFile to project.name)
     is Classycle -> listOf(reportFile to project.name)
     is FindBrokenInternalLinks -> listOf(reportFile.get().asFile to project.name)
     is DistributionTest -> listOf(
diff -ruN a/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/task-properties-validation.gradle.kts b/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/task-properties-validation.gradle.kts
--- a/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/task-properties-validation.gradle.kts	2026-05-01 03:47:57.296075282 +1000
+++ b/buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/task-properties-validation.gradle.kts	2026-05-01 03:47:57.300075250 +1000
@@ -19,13 +19,3 @@
     java
 }

-val validateTaskName = "validatePlugins"
-val reportFileName = "task-properties/report.txt"
-
-tasks.register<ValidatePlugins>(validateTaskName) {
-    val main = project.sourceSets.main.get()
-    classes.from(main.output)
-    classpath.from(main.runtimeClasspath)
-    outputFile.set(project.reporting.baseDirectory.file(reportFileName))
-    enableStricterValidation.set(true)
-}
