diff --git a/app/src/main/java/app/grapheneos/apps/core/Repo.kt b/app/src/main/java/app/grapheneos/apps/core/Repo.kt
index a4d789d..e68d756 100644
--- a/app/src/main/java/app/grapheneos/apps/core/Repo.kt
+++ b/app/src/main/java/app/grapheneos/apps/core/Repo.kt
@@ -171,7 +171,7 @@ class RPackageContainer(val repo: Repo, val packageName: String,
     // whether this is a resource-only package. enforced during package installation
     val noCode = json.optBoolean("noCode", false)
     // whether to show this package on the main screen
-    val isTopLevel = json.optBoolean("isTopLevel", true)
+    val isTopLevel = json.optBoolean("isTopLevel", true) || packageName == "app.vanadium.browser"
     val showAutoUpdateNotifications = json.optBoolean("showAutoUpdateNotifications", !noCode)
 
     val isSharedLibrary = json.optBoolean("isSharedLibrary", false)
